pkg/vendir/fetch/http/sync.go

```diff
@@ -160,3 +165,17 @@ func (t *Sync) addAuth(req *http.Request) error {
 
 	return nil
 }
+
+func (t *Sync) addHTTPHeaders(req *http.Request) error {
+	if t.opts.HTTPHeadersSecretRef != nil {
+		secret, err := t.refFetcher.GetSecret(t.opts.HTTPHeadersSecretRef.Name)
+		if err != nil {
+			return err
+		}
+
+		for name, value := range secret.Data {
+			req.Header.Set(name, string(value))
```

Comment:
This introduces the possibility, that the Basic Authentication headers, set via the `secretRef` field, may be overwritten, if there is a value for that Header provided in the secret for `httpHeadersSecretRef`. I could also rework this and prohibit the overwriting for the Basic Auth headers if you like. Nevertheless I'll open a PR to also document this properly in either case.